modules/packages/LinearAlgebra.chpl

```diff
@@ -2829,15 +2829,38 @@ module Sparse {
     }
     // matrix-matrix
     else if Adom.rank == 2 && Bdom.rank == 2 {
-      if !isCSArr(A) || !isCSArr(B) then
-        compilerError("Only CSR format is supported for sparse multiplication");
-      return _csrmatmatMult(A, B);
+      if !isCSArr(A) || !isCSArr(B) then {
+        if isCSArr(A) && !isCSArr(B) then {
+          return sparseDenseMatmul(A, B);
+        }
+        else if !isCSArr(A) && isCSArr(B) then {
+          var res = sparseDenseMatmul(B.T, A.T);
```

Comment:
As we discussed on the telecon, it might be good if `sparseDenseMatMul` handled this case without actually transposing the array. You might do it by passing an option to your routine. That might reduce the amount of copying back and forth.